usm_wrappers/salt_wrapper.py

```diff
@@ -63,6 +73,53 @@ def accept_minion(minion_id):
     return (True if out else False)
 
 
+def get_started_minions(minions=[], timeout=60):
+    minion_set = set(minions)
+    up_minions = set(runner.cmd('manage.up', []))
+    if minion_set:
+        started_minions = up_minions.intersection(minion_set)
+    else:
+        started_minions = up_minions
+
+    if not timeout:
+        return started_minions
+
+    time_spent = 0
+    while True:
+        wait = timeout - time_spent
+        if wait <= 0:
+            break
+        start_time = time.time()
+
+        ret = sevent.get_event(wait=wait, full=True, tag='salt/minion')
```

Comment:
The way get_event works is it waits for an event tag to maximum 'wait' time.  It breaks whenever an event available else it times out.  There is a possibility to get an event quickly than max 'wait' time.

I tried to start with constant period for 'wait' time which doesn't give any improvement.